src/c4v/classifier/classifier.py

```diff
@@ -0,0 +1,381 @@
+"""
+    This classifier class will perform basic experiments, receiving 
+    as arguments the training arguments and the columns to use from the training dataset
+"""
+# Python imports
+from typing             import Dict, List, Any, Tuple
+from pathlib            import Path
+from importlib          import resources
+from enum               import Enum
+# Third Party
+from pandas.core.frame  import DataFrame
+from transformers import (
+    RobertaTokenizer,
+    Trainer,
+    TrainingArguments,
+    RobertaForSequenceClassification,
+)
+from sklearn.metrics import accuracy_score, f1_score, recall_score, precision_score
+from sklearn.model_selection import train_test_split
+from datasets import Dataset
+import torch
+import pandas as pd
+import numpy as np
+import os
+
+from transformers.trainer_utils import EvalPrediction
+
+device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+
+BASE_C4V_FOLDER = os.path.join(os.environ.get("HOME"), "/.c4v")
+BASE_C4V_EXPERIMENTS_FOLDER = os.path.join(BASE_C4V_FOLDER, "/experiments")
+
+class Tags(Enum):
+    """
+        Possible tags variants for an article
+    """
+    DENUNCIA_FALTA_DEL_SERVICIO = "DENUNCIA FALTA DEL SERVICIO"
+
+class ClassifierExperiment:
```

Comment:
I think this class could be a Generic.

We can keep it as is for now to move forward but let me shed some light:

1. If we want to classify with more than one label (multi class instead of binary) the code is almost all the same.
2. If we want to use a different HF model, we just need to change the model,and the tokenizer (two methods)
3. This is for Text classification which could be reflected in the class name to keep the door open for image classification for example.
4. The text classification experiment could be a different class that consumes a text clasifier, which allows the experiment to define the dataset.